arangod/Replication2/StateMachines/Document/DocumentStateHandlersFactory.cpp

```diff
@@ -62,7 +65,8 @@ auto DocumentStateHandlersFactory::createSnapshotHandler(
   // instead, which returns a managed pointer.
   auto& ci = vocbase.server().getFeature<ClusterFeature>().clusterInfo();
   return std::make_shared<DocumentStateSnapshotHandler>(
-      std::make_unique<DatabaseSnapshotFactory>(vocbase), ci.rebootTracker());
+      std::make_unique<DatabaseSnapshotFactory>(vocbase), ci.rebootTracker(),
+      gid, createLogger(gid));
```

Comment:
I am a bit afraid of the order of evaluation of function arguments. Afaik C++ does not specify one, which makes it dependent on the calling convention.
Since I'm using `gid` twice in the function arguments, it was tricky to choose which one to move.
But I could create the loggerContext first and then move the gid.

Reply:
Fixed in https://github.com/arangodb/arangodb/pull/20012/commits/d5394d6c83c6ce73b3a8027e43145ee0381a616f